src/ORM/Association/DependentDeleteHelper.php

```diff
@@ -45,7 +45,11 @@ public function cascadeDelete(Association $association, EntityInterface $entity,
         /** @psalm-suppress InvalidArgument */
         $foreignKey = array_map([$association, 'aliasField'], (array)$association->getForeignKey());
         $bindingKey = (array)$association->getBindingKey();
-        $conditions = array_combine($foreignKey, $entity->extract($bindingKey));
+        $bindingValue = $entity->extract($bindingKey);
+        if (in_array(null, $bindingValue, true)) {
+            return true;
+        }
```

Comment:
Should this be if "any" or if "all" are null?

Reply:
Should this be an error? The delete cannot be safely cascaded and the intent is to have that done.

Reply:
Well, there is nothing to delete though. If you have an association based on a different field than the entity id, that could be null when deleting the entity.

Reply:
Can we delete the rows where the binding field is null then? There could be other uses for rows with non-null values, and the provided entity did have a null so an explicit match would be congruent with how we treat all the other values.

Reply:
I had that originally. You could end up deleting all rows in the target table where foreign key is null even though there was nothing attached to this entity. I wasn't sure which of these solutions was correct.

Reply:
I think we need to pick a side and either treat the null as we would all other values or raise an error as the operation is ambiguous and likely a mistake on the developer's behalf. Treating it as an error state is 'safer'. The intent of the current code is to do the delete but because of incorrect logic those deletes aren't happening.

Reply:
My guess is this is a bug and users don't use the feature enough to notice. I can imagine having an entity where the hasOne relationship is empty.

If an association is valid with a null foreign key, then we should delete nulls. If it isn't, and the property on the entity is left empty, then we should ignore it.

Reply:
Extended the test to show that a null binding key and null foreign key do create an association so the delete should not cascade.